tests/ui/rfcs/rfc-0000-never_patterns/macros.rs

```diff
@@ -0,0 +1,40 @@
+// run-pass
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+
+use ParsedAs::*;
+#[derive(Debug, PartialEq, Eq)]
+enum ParsedAs {
+    Pattern,
+    Never,
+    Other,
+}
+
+macro_rules! detect_pat {
+    ($p:pat) => {
```

Comment:
Wait, weren't the pat macro metavar matchers adjusted for editions >= 2021? Please make an `//edition:2021` version of this test.

Reply:
Yup, I checked on the playground, this is backwards-compatible, and in fact the assymetry of `! | true` vs `true | !` in edition 2021 can't be changed